### PR TITLE
TRUNK-5497: Updated the mysql-connector-java from 5.1.45 to 8.0.16, m…

### PR DESCRIPTION
**Pull request by reagan-meant**
…ysql-connector-mxj from 5.0.11 to 5.0.12 and mysql-connector-mxj-db-files from 5.0.11 to 5.0.12

Upgrade MySQL version From 5.7 to 8.0
OpenMRS CoreTRUNK-5497 Upgrade MySQL version From 5.7 to 8
Updated the mysql-connector-java from 5.1.45 to 8.0.16, mysql-connector-mxj from 5.0.11 to 5.0.12 and mysql-connector-mxj-db-files from 5.0.11 to 5.0.12"
https://issues.openmrs.org/browse/TRUNK-5497?jql=assignee%20in%20(reagan)%20ORDER%20BY%20created%20DESC

### CODE REVIEW

**Comment by ODORA0** [comment]
It would be cool @reagan-meant if you added ticket id as advised here https://wiki.openmrs.org/display/docs/Pull+Request+Tips. Good job though.

**Comment by mozzy11** [comment]
@reagan-meant  thanks for the work . 
Have you you seen @ODORA0 's comment?

**Comment by reagan-meant** [comment]
@Ruhanga everything is running perfectly..
@ODORA0 I hope it's fine now..

**Comment by mozzy11** [comment]
cc @dkayiwa .is this fine for merge??

**Comment by dkayiwa** [comment]
@reagan-meant did you say that everything is running perfectly using a real MySQL 8 database server?

**Comment by reagan-meant** [comment]
@dkayiwa  Yes...It was flawless

**Comment by dkayiwa** [comment]
@reagan-meant i do not believe so. Can you share a screenshot of your MySQL command line client showing the openmrs database and version of MySQL?

**Comment by mozzy11** [comment]
thats true @dkayiwa , thats why i also neeeded some other dev to confirm the core now supports mysql 8
see https://talk.openmrs.org/t/need-some-one-to-test-that-core-can-support-mysql-8/23604

**Comment by dkayiwa** [comment]
@mozzy11 am very sure that even the first dev did not run this on MySQL 8

**Comment by reagan-meant** [comment]
will share it shortly...i used my old machine...

**Comment by mozzy11** [comment]
> @mozzy11 am very sure that even the first dev did not run this on MySQL 8

Well the first dev says he ran it on mysql 8 , let him share the proof ,haha

**Comment by reagan-meant** [comment]
@dkayiwa @mozzy11  hope this evidence will suffice.....https://pastebin.com/t6LiRL8n

**Comment by dkayiwa** [comment]
@reagan-meant when you revert the changes in this pull request and compile again, do you get any errors while running openmrs?

**Comment by reagan-meant** [comment]
@dkayiwa when i compile on the master without these changes.I get this error at initialsetup after starting the server https://pastebin.com/c84kwpF9

**Comment by mozzy11** [comment]
@dkayiwa , first dev gives un-mistakable evidence , haha

File: pom.xml [hunk 1]
```diff
@@ -398,16 +398,19 @@
 				<artifactId>commons-fileupload</artifactId>
 				<version>1.3.3</version>
 			</dependency>
+				<!-- https://mvnrepository.com/artifact/mysql/mysql-connector-java -->
```
Comment (dkayiwa) [inline pom.xml]: What is the use of the above comment?

File: pom.xml [hunk 2]
```diff
@@ -398,16 +398,19 @@
...
 			<dependency>
 				<groupId>mysql</groupId>
 				<artifactId>mysql-connector-java</artifactId>
-				<version>5.1.45</version>
+				<version>8.0.16</version>
 				<scope>runtime</scope>
 			</dependency>
+			<!-- https://mvnrepository.com/artifact/mysql/mysql-connector-mxj -->
```
Comment (dkayiwa) [inline pom.xml]: What is the use of the above comment?

File: webapp/pom.xml [hunk 3]
```diff
@@ -152,25 +152,31 @@
...
-                </dependency>
+               
 
-                <dependency>
-                    <groupId>com.mysql</groupId>
-                    <artifactId>mysql-connector-mxj</artifactId>
-                    <version>5.0.11</version>
-                    <scope>compile</scope>
-                </dependency>
+				<!-- https://mvnrepository.com/artifact/mysql/mysql-connector-java -->
```
Comment (dkayiwa) [inline webapp/pom.xml]: What is the use of the above comment?

File: webapp/pom.xml [hunk 4]
```diff
@@ -152,25 +152,31 @@
...
+				<dependency>
+					<groupId>mysql</groupId>
+					<artifactId>mysql-connector-java</artifactId>
+					<version>8.0.16</version>
+					<scope>runtime</scope>
+				</dependency>
+				<!-- https://mvnrepository.com/artifact/mysql/mysql-connector-mxj -->
```
Comment (dkayiwa) [inline webapp/pom.xml]: What is the use of the above comment?

File: webapp/pom.xml [hunk 5]
```diff
@@ -152,25 +152,31 @@
...
+				<dependency>
+					<groupId>mysql</groupId>
+					<artifactId>mysql-connector-mxj</artifactId>
+					<version>5.0.12</version>
+					<scope>compile</scope>
+				</dependency>
+
+				<!-- https://mvnrepository.com/artifact/mysql/mysql-connector-mxj-db-files -->
```
Comment (dkayiwa) [inline webapp/pom.xml]: What is the use of the above comment?

**Comment by dkayiwa** [comment]
@reagan-meant can you test your changes on MySQL 5.7 and see if you are able to run the openmrs setup wizard successfully up to the end?

File: pom.xml (hunk 1 above)
Comment (reagan-meant) [reply pom.xml]: It was part of the comments in maven...I will remove them and push again

**Comment by reagan-meant** [comment]
i am runing into the following errors after (creating the db, creating tables,inserting some demo data but fails to update the data) and throws this

**Comment by dkayiwa** [comment]
So can you look into making it backwards compatible?

**Comment by spaffrath** [comment]
Hello, Looking at the changes in the webapp pom.xml file got me thinking that the "integration-test" profile which is edited must not be being used. Otherwise, it would have been throwing errors from the misspelled dependencies. Does anyone know this for certain? It may be best to remove that profile completely if it hasn't been used in 8 years (added in 2011).

Secondly, if there is a reason to keep that profile then it is best to let the dependencies inherit their versions from the parent pom.xml file so they don't need to be managed. I am new to the project so please correct me if I'm wrong but I thought I'd share these observations.

**Pull request closed without merging**